capture/engine_scripts/puppet/loadCookies.js

```diff
@@ -1,22 +1,22 @@
 const fs = require('fs');
 
 module.exports = async (page, scenario) => {
-  let cookies = [];
+  const cookies = [];
   const cookiePath = scenario.cookiePath;
 
   // READ COOKIES FROM FILE IF EXISTS
   if (fs.existsSync(cookiePath)) {
     cookies = JSON.parse(fs.readFileSync(cookiePath));
   }
 
-  // MUNGE COOKIE DOMAIN
+  // Inject cookie domain, url, secure from scenario url.
+  const url = new URL(scenario.url);
   cookies = cookies.map(cookie => {
-    if (cookie.domain.startsWith('http://') || cookie.domain.startsWith('https://')) {
-      cookie.url = cookie.domain;
-    } else {
-      cookie.url = 'https://' + cookie.domain;
+    cookie.url = `${url.protocol}//${url.hostname}`;
```

Comment:
I prefer this way more to what is currently here.  I am a little conflicted about using the "magic" hostname since this could differ from the one exported in the cookie-- but I think this is your whole point.  So, I am supportive of this because it will probably work more often than not.